src/realm/sync/noinst/client_impl_base.cpp

```diff
@@ -2063,21 +2066,21 @@ void Session::receive_download_message(const SyncProgress& progress, std::uint_f
         }
     }
 
-    receive_download_message_hook();
+    receive_download_message_hook(progress, query_version, batch_state);
 
-    if (batch_state == DownloadBatchState::LastInBatch) {
-        update_progress(progress); // Throws
+    if (process_flx_bootstrap_message(progress, batch_state, query_version, received_changesets)) {
```

Comment:
shouldn't be pass `downloadable_bytes` and save it in the metadata table too?

Reply:
`downloadable_bytes` doesn't really have meaning in FLX sync and this should always be zero.